examples/jsm/renderers/webgpu/nodes/WebGPUNodeBuilder.js

```diff
@@ -55,12 +69,52 @@ class WebGPUNodeBuilder extends NodeBuilder {
 
 				this.addSlot( 'fragment', new NodeSlot( material.colorNode, 'COLOR', 'vec4' ) );
 
+			} else {
+
+				this.addSlot( 'fragment', new NodeSlot( new MaterialNode( MaterialNode.COLOR ), 'COLOR', 'vec4' ) );
```

Comment:
So when the application does not assign a custom node to `material.colorNode`, `MaterialNode` is used to map common material properties like `color` or `map` to the slot, right.

When I understand it correctly, `MaterialReferenceNode` has the purpose to connect a material's property with a property of a node class, right`? One could say `ReferenceNode` is the glue that assigns an object property to a node class property, correct?

Reply:
That's right.

Reply:
And `MaterialNode` connects the `inputs` for a composed output.